### PR TITLE
[Merged by Bors] - Eisenstein series are modular forms

### PR DESCRIPTION
**Pull request by CBirkbeck**
We show Eisenstein series are modular forms.

---

- [x] depends on: #12456
[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/ModularForms/EisensteinSeries/ModularForm.lean [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# Eisenstein series are Modular Forms
+
+We show that Eisenstein series of weight `k` and level `Γ(N)` with congruence condition
+`a : Fin 2 → ZMod N` are Modular Forms.
+-/
+
+noncomputable section
+
+
+namespace EisensteinSeries
```
Comment (riccardobrasca) [inline Mathlib/NumberTheory/ModularForms/EisensteinSeries/ModularForm.lean:21]:
```suggestion
noncomputable section

namespace EisensteinSeries
```

File: Mathlib/NumberTheory/ModularForms/EisensteinSeries/ModularForm.lean [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+/-- This defines Eisenstein series as modular forms of weight `k`, level `Γ(N)` and congruence
+condition given by `a: Fin 2 → ZMod N`. -/
+def eisensteinSeries_MF {k : ℤ} {N : ℕ+} (hk : 3 ≤ k) (a : Fin 2 → ZMod N) :
```
Comment (riccardobrasca) [inline Mathlib/NumberTheory/ModularForms/EisensteinSeries/ModularForm.lean:25]: The name `EisensteinSeries.eisensteinSeries_MF` seems weird. Maybe `ModularForms.eisensteinSeries`?

**Review by riccardobrasca — APPROVED** [review]
Can you add a TODO to prove that it is not zero? Thanks!

bors d+

File: Mathlib/NumberTheory/ModularForms/EisensteinSeries/ModularForm.lean (hunk 2 above)
Comment (CBirkbeck) [reply Mathlib/NumberTheory/ModularForms/EisensteinSeries/ModularForm.lean:25]: Hmm good point, we actually already have `EinsesteinSeries.eisensteinSeries` which seems like a bad thing. I might put this in the root namespace and then make the other one `ModularForms.eisensteinSeries_MF` to denote its a mod form. Does that sound ok?

**Comment by grunweg** [comment]
> As for the TODO, I already have this, but its waiting on the q-expansion stuff thats coming. Should that still be a todo if its in the process of being PRd?

I would do so - the next PR can remove the TODO. If that other PR doesn't land soon (for whatever reason), the current file is in a more useful state.

File: Mathlib/NumberTheory/ModularForms/EisensteinSeries/ModularForm.lean (hunk 2 above)
Comment (riccardobrasca) [reply Mathlib/NumberTheory/ModularForms/EisensteinSeries/ModularForm.lean:30]: Yes, no problem.

**Pull request closed without merging**